### ISSUE TITLE
CPU bindings

### ISSUE THREAD

**Issue by cloudspins** [dwavesystems/dwave-pytorch-plugin#15]
https://github.com/dwavesystems/dwave-pytorch-plugin/blob/a921b19b304a1a8d293d472cdf311243faa08706/dwave/plugins/torch/boltzmann_machine.py#L239

Gradient transforms from Bolzmann machine Vectors without Torch could be more useful than integrating the Torch API which is known to be redundant and noisy in many area's. Basically reducing and transforming the vector to a restricted Bolzmann Machine and representing the data with Vector, Matrix (sparse, dense), Decode and Encode. Also, the Use Case can be specified more easily without Torch; "basic image processing" with QUBO and QSVM for example, or even a Use Case of "simulation of gases" and discovery of parameters.

With for example, TensorFlow; one could potentially map graph parameters onto a D-Wave Graph in a more meaningful manner. Of course the Ising (or other) model must then accurately correspond with the Tensor model. 

Below an example in psuedocode:

```python
import networkx as nx 
import random ... 
...
G[i] = nx.generators.complete_graph(Tensor.graph,Tensor.Size) 
...
for node in G[i].nodes: ...  #just an example, construct datastructure
   G[i].nodes[node]['bias'] = random.choice([1,-1])
 for edge in G[i].edges: ... #just an example, use actual TensorFlow to map
   G[i].edges[edge]['quadratic'] = random.choice([1,-1]) ... 
select bqm = dimod.from_networkx_graph(
G, ... vartype='REAL', ... edge_attribute_name='quadratic') #for the type values
from bqm_sub = dimod.from_networkx_graph( #for the type selections
vartype='SPIN', ...
```

Additional considerations must be the "reinforcement learning" and "classification mapping" (i.e. Transformers) method for correctly processing the qubit spin configurations in approximating the free energy of the quantum Boltzmann machine (QBM). This could then potentially lead to a hybrid (quantum) Maxwell-Boltzmann (input), Fermi-Diraq (zero-state  statistics) and Bose-Einstein (output) machine (QEM), if such a machine has not already been made. From the figure it can be observed that such an approach would yield different types of results, which can be visualized with the "spin states" using D-Wave's visualization tools.

![Image](https://github.com/user-attachments/assets/1648f3d4-28aa-4bce-bdc1-5ac075783d21)

.
